icu4c/source/common/unicode/char16ptr.h

```diff
@@ -306,6 +307,36 @@ inline OldUChar *toOldUCharPtr(char16_t *p) {
     return reinterpret_cast<OldUChar *>(p);
 }
 
+#ifndef U_FORCE_HIDE_INTERNAL_API
+/**
+ * Is T convertible to a std::u16string_view or to a 16-bit std::wstring_view?
+ * @internal
+ */
+template<typename T>
+constexpr bool ConvertibleToU16StringView =
+    std::is_convertible_v<T, std::u16string_view> ||
+    (U_SIZEOF_WCHAR_T==2 && std::is_convertible_v<T, std::wstring_view>);
+
+namespace internal {
+/**
+ * Pass-through overload.
+ * @internal
+ */
+inline std::u16string_view toU16StringView(std::u16string_view sv) { return sv; }
```

Comment:
What's this doing?  It looks like a no-op.

Reply:
It is, it’s just there so we can call toU16StringView on something that may or may not be a u16string_view already in the templates.
(We could also do that with `if constexpr` but that seems overkill.)